build/build-mrt.yml

```diff
@@ -118,121 +118,122 @@ steps:
-    testRunTitle: 'test MRT: UnpackagedTests - $(buildPlatform)'
-    platform: '$(buildPlatform)'
-    configuration: '$(buildConfiguration)'
-
-# Copy and Publish the test binaries so we can run them in Helix later.
-- task: CopyFiles@2
-  displayName: 'copy mrtcore test binaries'
-  condition: and(succeeded(), or(eq(variables['buildPlatform'], 'x86'), eq(variables['buildPlatform'], 'x64')))
-  inputs:
-    SourceFolder: '${{ parameters.MRTBinariesDirectory }}\Release\$(buildPlatform)'
-    Contents: |
-      MrmBaseUnitTests\**
-      MrmUnitTest\**
-      MrtCoreManagedTest\**
-      MrtCoreUnpackagedTests\**
-    TargetFolder: '$(Agent.TempDirectory)\mrt_testbin\$(buildPlatform)'
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish mrtcore test binaries'
-  condition: and(succeeded(), or(eq(variables['buildPlatform'], 'x86'), eq(variables['buildPlatform'], 'x64')))
-  inputs:
-    PathtoPublish: '$(Agent.TempDirectory)\mrt_testbin\$(buildPlatform)'
-    artifactName: mrtcore_test_binaries_$(buildPlatform)
-
-- task: ComponentGovernanceComponentDetection@0
-  inputs:
-    scanType: 'Register'
-    verbosity: 'Verbose'
-    alertWarningLevel: 'Medium'
-    failOnAlert: true
-
-- task: CopyFiles@2
-  displayName: 'copy binaries for signing'
-  inputs:
-    SourceFolder: '${{ parameters.MRTBinariesDirectory }}\Release\$(buildPlatform)'
-    Contents: |
-      mrm\mrm.dll
-      mrm\mrm.lib
-      mrm\mrm.pdb
-      Microsoft.ApplicationModel.Resources\Microsoft.ApplicationModel.Resources.dll
-      Microsoft.ApplicationModel.Resources\Microsoft.ApplicationModel.Resources.pdb
-    TargetFolder: '$(Build.ArtifactStagingDirectory)\mrt_raw\lib\$(buildPlatform)'
-    flattenFolders: true
-
-- task: CopyFiles@2
-  displayName: 'copy projection binaries and symbols for signing'
-  condition: and(succeeded(), eq(variables['buildPlatform'], 'x86'))
-  inputs:
-    SourceFolder: '${{ parameters.MRTBinariesDirectory }}\Release\$(buildPlatform)\Microsoft.ApplicationModel.Resources.Projection'
-    Contents: |
-      Microsoft.ApplicationModel.Resources.Projection.dll
-      Microsoft.ApplicationModel.Resources.Projection.pdb
-    TargetFolder: '$(Build.ArtifactStagingDirectory)\mrt_raw\lib\anycpu\net5.0'
-    flattenFolders: false
-
-- task: CopyFiles@2
-  displayName: 'copy winmd for signing'
-  condition: and(succeeded(), eq(variables['buildPlatform'], 'x86'))
-  inputs:
-    SourceFolder: '${{ parameters.MRTBinariesDirectory }}\Release\$(buildPlatform)\Microsoft.ApplicationModel.Resources'
-    Contents: |
-      Microsoft.ApplicationModel.Resources.winmd
-    TargetFolder: '$(Build.ArtifactStagingDirectory)\mrt_raw\lib\anycpu'
-    flattenFolders: true
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish Artifact: Binaries'
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)\mrt_raw'
-    ArtifactName: 'mrtcore_binaries'
+- ${{ if eq(variables.buildOnly, 'false') }}:
```

Comment:
why this check?

Reply:
because the copyto steps afterwards fails for various reasons

Reply:
is that because we hardcoded "Release" instead of using $(buildConfiguration)? we can skip copy, but we want to have test run. At the very minimum, we want to have the binlog published so that we have more information why build failed.

Reply:
I think the test runs relies on the copto tasks